src/OpenTelemetry.Instrumentation.ConfluentKafka/InstrumentedConsumer.cs

```diff
@@ -48,56 +48,37 @@ public void SetSaslCredentials(string username, string password)
         this.consumer.SetSaslCredentials(username, password);
     }
 
-    public ConsumeResult<TKey, TValue>? Consume(int millisecondsTimeout)
-    {
-        var start = DateTimeOffset.UtcNow;
-        ConsumeResult<TKey, TValue>? result = null;
-        ConsumeResult consumeResult = default;
-        string? errorType = null;
-        try
-        {
-            result = this.consumer.Consume(millisecondsTimeout);
-            consumeResult = ExtractConsumeResult(result);
-            return result;
-        }
-        catch (ConsumeException e)
-        {
-            (consumeResult, errorType) = ExtractConsumeResult(e);
-            throw;
-        }
-        finally
-        {
-            var end = DateTimeOffset.UtcNow;
-            if (result is { IsPartitionEOF: false })
-            {
-                this.InstrumentConsumption(start, end, consumeResult, errorType);
-            }
-        }
-    }
+    public ConsumeResult<TKey, TValue>? Consume(int millisecondsTimeout) =>
+        this.Consume(TimeSpan.FromMilliseconds(millisecondsTimeout));
 
     public ConsumeResult<TKey, TValue>? Consume(CancellationToken cancellationToken = default)
     {
         var start = DateTimeOffset.UtcNow;
         ConsumeResult<TKey, TValue>? result = null;
         ConsumeResult consumeResult = default;
         string? errorType = null;
+        using var activity = this.StartReceiveActivity();
         try
         {
             result = this.consumer.Consume(cancellationToken);
             consumeResult = ExtractConsumeResult(result);
+            UpdateActivityWithConsumeResults(activity, result);
             return result;
         }
         catch (ConsumeException e)
         {
             (consumeResult, errorType) = ExtractConsumeResult(e);
+            activity?.AddException(e);
+            activity?.SetStatus(ActivityStatusCode.Error);
```

Comment:
a common complaint on Azure messaging that might be applicable here is that we record timeout as error, while all it's saying that someone polled and there were no messages on the topic to consume (which is absolutely fine).

do you know how it manifests on Kafka .NET? Would we record this as error?